### PR TITLE
feat: support env filter 

### LINKED ISSUE #54: EnvFilter
Support loading env var filters and like https://crates.io/crates/env_logger.

### PR DESCRIPTION
**Pull request by tisonkun**
This closes #54

### CODE REVIEW

File: examples/env_filter.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use logforth::append;
+use logforth::filter::EnvFilter;
+use logforth::layout::TextLayout;
+use logforth::Dispatch;
+use logforth::Logger;
+
+fn main() {
```
Comment (andylokandy) [inline examples/env_filter.rs:21]: It may be better if you add some command line example for different env filter cases.

File: src/logger.rs [hunk 2]
```diff
@@ -89,7 +89,7 @@ impl<const LAYOUT: bool, const APPEND: bool> Dispatch<LAYOUT, APPEND> {
 impl Dispatch {
     fn enabled(&self, metadata: &Metadata) -> bool {
```
Comment (andylokandy) [inline src/logger.rs:90]: This is function unused?

**Review by andylokandy — APPROVED** [review]

File: src/logger.rs (hunk 2 above)
Comment (tisonkun) [reply src/logger.rs:90]: It's used in L175

File: examples/env_filter.rs (hunk 1 above)
Comment (tisonkun) [reply examples/env_filter.rs:21]: Things like `RUST_LOG=info cargo run --features="env-filter" --example env_filter` ..

I don't find a place suitable for this and would prefer to postpone it as a follow-up.

**Pull request merged**